sys/ztimer/auto_init.c

```diff
@@ -128,19 +125,17 @@
 #      define INIT_ZTIMER_RTT 1
 #    endif
 #    define ZTIMER_SEC_CONVERT_LOWER_FREQ ZTIMER_RTT_FREQ
-#  else
-#    ifdef ZTIMER_RTC
-#      define ZTIMER_SEC_RTC
-#      ifndef INIT_ZTIMER_RTC
-#        define INIT_ZTIMER_RTC 1
-#      endif
-#    else
-#      define ZTIMER_SEC_TIMER
-#      ifndef INIT_ZTIMER_TIMER
-#        define INIT_ZTIMER_TIMER 1
-#      endif
-#      define ZTIMER_SEC_CONVERT_LOWER_FREQ ZTIMER_TIMER_FREQ
+#  elif defined(ZTIMER_RTC)
+#    define ZTIMER_SEC_RTC
+#    ifndef INIT_ZTIMER_RTC
+#      define INIT_ZTIMER_RTC 1
+#    endif
+#  elif defined(ZTIMER_TIMER)
+#    define ZTIMER_SEC_TIMER
+#    ifndef INIT_ZTIMER_TIMER
+#      define INIT_ZTIMER_TIMER 1
 #    endif
+#    define ZTIMER_SEC_CONVERT_LOWER_FREQ ZTIMER_TIMER_FREQ
 #  endif
 #endif
```

Comment:
```suggestion
#      define INIT_ZTIMER_TIMER 1
#    endif
#    define ZTIMER_SEC_CONVERT_LOWER_FREQ ZTIMER_TIMER_FREQ
#  else
#    error "No backend available to implement ZTIMER_SEC"
#  endif
#endif
```

?

Reply:
this error message is in l 207 ante commit l 202 post commit

in step 1 a backend is selected (step 1 has all the precedence logic)

if non is (could be) selected a error message is printed in step 3

we can have extra error message in step one but we don't need them there